src/rubrix/client/datasets.py

```diff
@@ -486,6 +490,8 @@ def entities_to_dict(
                 ]
             elif key == "id":
                 ds_dict[key] = [str(rec.id) for rec in self._records]
+            elif key == "metadata":
+                ds_dict[key] = [getattr(rec, key) or None for rec in self._records]
```

Comment:
This line will replace the empty `metadata` to a `None` because HF does not support empty dict as value?

In that case, could we just define the `metadata` field as optional? 

What about with other fields like metrics. Should we prevent empty dict in a more general way?

What should be the behaviour with following code?:
```python
dataset = rb.DatasetForText2Text([rb.Text2TextRecord(text="mock", metrics={})])
hf_dataset = dataset.to_datasets()
hf_dataset.push_to_hub("dataset_with_empty_metrics")
```

Reply:
As an additional comment, we could improve this code by swapping the for loops:

```python
ds_dict = defaultdict(list)
for rec in self._records:
   for key in self._RECORD_TYPE.__fields__:
      ds_dict[key].append(....)
```